converters/volAddNoise.cpp

```diff
@@ -0,0 +1,121 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+/**
+ * @file imgAddNoise
+ * @ingroup Examples
+ * @author David Coeurjolly (\c david.coeurjolly@liris.cnrs.fr)
+ *
+ * @date 2015/03/24
+ *
+ *
+ * This file is part of the DGtal library.
+ */
+
+
+#include <DGtal/base/Common.h>
+#include "DGtal/io/readers/GenericReader.h"
+#include "DGtal/io/writers/GenericWriter.h"
+#include "DGtal/images/ImageContainerBySTLVector.h"
+#include "DGtal/images/ImageSelector.h"
+#include <DGtal/geometry/volumes/KanungoNoise.h>
+#include <DGtal/images/IntervalForegroundPredicate.h>
+#include <boost/program_options/options_description.hpp>
+#include <boost/program_options/parsers.hpp>
+#include <boost/program_options/variables_map.hpp>
+
+#include <vector>
+#include <string>
+#include <climits>
+
+ 
+using namespace DGtal;
+
+///////////////////////////////////////////////////////////////////////////////
+namespace po = boost::program_options;
+
+/**
+ * Missing parameter error message.
+ *
+ * @param param
+ */
+void missingParam ( std::string param )
+{
+  trace.error() <<" Parameter: "<<param<<" is required..";
+  trace.info() <<std::endl;
+  exit ( 1 );
+}
+
+typedef ImageSelector < Z3i::Domain, unsigned char>::Type MyImage;
+
+
+int main( int argc, char** argv )
+{
+  // parse command line ----------------------------------------------
+  po::options_description general_opt("Allowed options are: ");
+  general_opt.add_options()
+    ("help,h", "display this message")
+    ("input,i", po::value<std::string>(), "input image file name (any 3D image format accepted by DGtal::GenericReader)")
```

Comment:
In 3D perhaps it could be interesting to add the threshold min/max for the predicate ?